met-api/src/met_api/services/comment_service.py

```diff
@@ -64,20 +64,19 @@ def extract_comments(cls, survey_submission: SubmissionSchema, survey: SurveySch
         components = list(survey_form.get('components', []))
         if len(components) == 0:
             return []
-
         # get the 'key' for each component that has 'inputType' text and filter out the rest.
         comments_keys = [
             component.get('key', None) for component in components
             if component.get('inputType', None) == 'text']
-
         submission = survey_submission.get('submission_json', {})
         comments_texts = [submission.get(key, None) for key in comments_keys]
```

Comment:
I'm gonna throw a different approach and leave it to Jad to decide, but when assigning comments_texts you can filter out the blank ones something like:  
        comments_texts = [submission.get(key, None) for key in comments_keys
        if submission.get(key, None) != '']

Reply:
yup this is definitely cleaner and more performant

Reply:
good idea